### PR TITLE
federation: Move k/test/e2e_federation package to k/federation/test/e2e

### PR DESCRIPTION
**Pull request by shashidharatd**
No functional change. Just move federation e2e tests inside k/federation directory, so that it is easy to move federation out of core.

Issue ref: #52992

**Release note**:
```release-note
NONE
```

/cc @kubernetes/sig-federation-pr-reviews @irfanurrehman @madhusudancs

### CODE REVIEW

**Comment by irfanurrehman** [comment]
@shashidharatd can you please also consider moving `test/integration/federation` inside `federation`, as part of this PR or a separate PR.
Thanks!

**Comment by shashidharatd** [comment]
@irfanurrehman, i have moved federation integration tests into the federation directory as you suggested and all tests seems to be fine. request to PTAL and give lgtm. Thanks

**Comment by shashidharatd** [comment]
@csbell, request to PTAL and approve this PR asap, as this involves the moving out code to another path, there is potential for lot of conflicts as other PR's gets merged. Request to please lgtm and approve at the earliest. Thanks !

File: test/e2e/e2e.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	runtimeutils "k8s.io/apimachinery/pkg/util/runtime"
 	clientset "k8s.io/client-go/kubernetes"
+	federationtest "k8s.io/kubernetes/federation/test/e2e"
```
Comment (madhusudancs) [inline test/e2e/e2e.go]: Do you plan to remove this in the future?

File: hack/make-rules/test-integration.sh [hunk 2]
```diff
@@ -49,6 +49,9 @@ kube::test::find_integration_test_dirs() {
     find vendor/k8s.io/apiextensions-apiserver/test/integration/ -name '*_test.go' -print0 \
       | xargs -0n1 dirname | sed "s|^|${KUBE_GO_PACKAGE}/|" \
       | LC_ALL=C sort -u
+    find federation/test/integration/ -name '*_test.go' -print0 \
```
Comment (madhusudancs) [inline hack/make-rules/test-integration.sh]: You plan to remove this (or move this to some other place) in the future?

**Review by madhusudancs — COMMENTED** [review]
/lgtm
/approve

File: test/e2e/e2e.go (hunk 1 above)
Comment (shashidharatd) [reply test/e2e/e2e.go]: yes, we plan to remove this from k/k and we shall do it as soon as we build the ci jobs on k/f and testing works from there.

File: hack/make-rules/test-integration.sh (hunk 2 above)
Comment (shashidharatd) [reply hack/make-rules/test-integration.sh]: yes, same answer as above comment.
Thanks @madhusudancs

**Pull request merged**